template/script.sh.erb

```diff
@@ -1,38 +1,62 @@
 #!/usr/bin/env bash
-<%-
-    # Ensure that code-server always starts up in either a user defined directory or the home directory
-    working_dir = Pathname.new(context.working_dir)
-    if ! working_dir.exist? || working_dir.to_s.empty?
-        working_dir = Pathname.new(ENV['HOME'])
-    elsif working_dir.file?
-        working_dir = working_dir.parent
-    end
+<%
+require 'shellwords'
+
+# Determine the application root from session token.
+app_root = BatchConnect::App.from_token(session.token).root.realpath
+
+# Escape working directory from user input string.
+escaped_working_dir = Shellwords.escape(context.working_dir)
```

Comment:
Do you have an example string that would be problematic requiring `Shellwords.escape` prior to using in `Pathname.new`?

Reply:
`context.working_dir` comes from the user. You recommended to use `Shellwords.escape` a while back

I just tested this with a random input like `/\asdf/\\/asdf/\` without using `Shellwords.escape` and the outcome was the same: since that directory doesn't exist, the working directory defaults to `working_dir = Pathname.new(ENV['HOME'])`

`Shellwords.escape` isn't relevant here, good catch